src/GraphNavigator/DeserializationGraphNavigator.php

```diff
@@ -197,7 +214,10 @@ public function accept($data, ?array $type = null)
                     $this->context->pushPropertyMetadata($propertyMetadata);
                     try {
                         $v = $this->visitor->visitProperty($propertyMetadata, $data);
-                        $this->accessor->setValue($object, $v, $propertyMetadata, $this->context);
+
+                        if (null !== $v || true === $this->shouldDeserializeNull) {
```

Comment:
so here is where you prevent it instead of doing in the visitor

Reply:
This is what is breaking the tests and now that I think about it, it is not needed at all. Without it, tests pass.